src/nodes.coffee

```diff
@@ -2871,19 +2870,17 @@ exports.StringWithInterpolations = class StringWithInterpolations extends Base
               match
         fragments.push @makeCode value
       else
-        fragments.push @makeCode '$' if not @csx
+        fragments.push @makeCode '$' unless @csx
         code = element.compileToFragments(o, LEVEL_PAREN)
-        code = @wrapInBraces code if not @isNestedTag element
+        code = @wrapInBraces code unless @isNestedTag element
         fragments.push code...
-    fragments.push @makeCode '`' if not @csx
+    fragments.push @makeCode '`' unless @csx
     fragments
 
   isNestedTag: (element) ->
-    @csx and
-    (exprs = element?.body?.expressions) and
-    exprs.length is 1 and
-    (call = exprs?[0]) instanceof Call and
-    call.csx
+    exprs = element?.body?.expressions
+    call = exprs?[0]
+    @csx and exprs and exprs.length is 1 and call instanceof Call and call.csx
```

Comment:
Ah okay, we can put `return unless @csx` at the top. What I was trying to achieve was to move the assignments out of the conditionals. As a general style rule I’m trying to avoid assignments inside conditionals (like `if (a = 1)`).